frontend/src/lib/components/Cards/TextCard/TextCard.tsx

```diff
@@ -26,21 +26,27 @@ interface TextCardProps extends React.HTMLAttributes<HTMLDivElement>, Resizeable
     showEditingControls?: boolean
 }
 
-interface TextCardBodyProps extends Pick<React.HTMLAttributes<HTMLDivElement>, 'style'> {
+interface TextCardBodyProps extends Pick<React.HTMLAttributes<HTMLDivElement>, 'style' | 'className'> {
     text: string
     closeDetails?: () => void
 }
 
-export function TextCardBody({ text, closeDetails, style }: TextCardBodyProps): JSX.Element {
-    useEffect(() => {
-        window.dispatchEvent(new Event('resize'))
-    }, [style?.height])
+export function TextCardBody({ text, closeDetails, style, className }: TextCardBodyProps): JSX.Element {
+    const { fontSize, ref } = useFitText({
+        maxFontSize: 200,
+        resolution: 5,
+        text,
+    })
```

Comment:
Sorry I didn't get to this sooner like I said, fell into some rabbit holes…
Now, I haven't look that much into the text fitting code yet – after playing with this, I think there is a slightly more burning product/design question here:
Why not instead make the typography of these text cards nicer, and allow users to use headings and body text as _they_ see fit?

For example, this Markdown:

```markdown
# Lorem ipsum
## dolor sit amet
### consectetur adipisicing elit
Proin nibh augue, suscipit a, scelerisque sed, lacinia in, mi.
```

…looks like this in GitHub:

> # Lorem ipsum
> ## dolor sit amet
> ### consectetur adipisicing elit
> Proin nibh augue, suscipit a, scelerisque sed, lacinia in, mi.

There is an intuitive hierarchy, and the details (font sizes, weights, margins) look nice enough.

Here's what this looks like in this PR:

<img width="813" alt="md" src="https://user-images.githubusercontent.com/4550621/203138338-5e4b9971-a1f8-49b2-9c3e-43d215d44224.png">

Besides the first line being mostly obscured, there is the issue of body text being barely less prominent than the top-level heading, and the spacing somewhat too lax. So this body text isn't really fit for longer forms of text, and the headings don't do a great job of standing out. Also, fitting font size individually for each card will result in them having varying typography, which is not a great look. Overall this is more of a design problem and less an engineering one.

Looking at this PR vs `master` and taking the above considerations into account, I'd rather suggest making text card font size responsive based on viewport size (ensuring consistent typography within a device) + some tuning the current h1/h2/p etc. styling within `TextCard`.

Reply:
🤔 totally agree. lets split this up then

#12933 to fix the preview